### DISCUSSION TITLE
编辑器是否能通过插件实现类似 vim 模式等编辑模式？

### DISCUSSION THREAD

**Discussion by charlestang** [imzbf/md-editor-v3#386] [Q&A]
编辑器是否能通过插件实现类似 vim 模式等编辑模式？

**Comment by imzbf** [comment]
具体的是指？

**Reply by charlestang** [reply]
就是在编辑器里书写，像 vim 那样可以进入 normal 模式，insert 模式，具体可以看看 vscode 的 vim插件的效果。

**Reply by charlestang** [reply]
我看 codemirror 似乎可以实现 https://codemirror.net/3/demo/vim.html 参考这个 demo

**Reply by imzbf** [reply]
https://github.com/replit/codemirror-vim#usage

```js
import { config } from 'md-editor-v3';
import { vim } from '@replit/codemirror-vim';

config({
  codeMirrorExtensions(theme, extensions) {
    return [...extensions, lineNumbers(), vim()];
  }
})
```

这倒是可以，不过应该挺多细节不一样的，像样式、快捷键等等

**Reply by charlestang** [reply]
谢谢，我去试试效果好不好

**Reply by charlestang** [reply]
我实验了一下，大体上还行，不过，backspace 和 enter，竟然可以无视模式直接输入字符，另外还有显示命令的 panel，确实样式不对，颜色全黑，看不见。另外 :w 这样的命令，是没有handler 的。还得慢慢琢磨，不过光标移动，模式切换大体上有了。
